src/metadata.rs

```diff
@@ -47,25 +65,54 @@ pub trait Metadata: MajorObject {
         Ok(_string(c_res))
     }
 
+    /// Metadata in GDAL is partitioned into namespaces, knows as "domains" in the
+    /// [GDAL Data Model](https://gdal.org/user/raster_data_model.html#metadata).
+    /// GDAL types with metadata (a.k.a. "Major Objects") have a default or "root" domain
+    /// identified by the empty string (`""`).
+    /// Specific "Major Object" types may have other conventionally recognized domains.
+    /// For example, in raster [`Dataset`][crate::Dataset]s you may come across the domains
+    /// `SUBDATASETS`, `IMAGE_STRUCTURE`, `RPC`, `IMAGERY`, `xml:`, etc.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use gdal::{Dataset, Metadata};
+    /// # fn main() -> gdal::errors::Result<()> {
+    /// let dataset = Dataset::open("fixtures/labels.tif")?;
+    /// assert!(dataset.metadata_domains().contains(&"IMAGE_STRUCTURE".to_string()));
+    /// # Ok(())
+    /// # }
+    /// ```
     fn metadata_domains(&self) -> Vec<String> {
         let mut domains = Vec::new();
         let c_res = unsafe { gdal_sys::GDALGetMetadataDomainList(self.gdal_object_ptr()) };
 
         if !c_res.is_null() {
-            for i in 0.. {
-                let p = unsafe { *c_res.offset(i) };
-                if p.is_null() {
-                    break;
-                }
-
-                domains.push(_string(p));
-            }
+            domains.append(&mut _string_array(c_res));
         }
         unsafe { gdal_sys::CSLDestroy(c_res) };
 
         domains
     }
 
+    /// Get all the metadata values within the given `domain`. Returns `None` if
+    /// domain is not defined.
+    /// Entries in the returned `Vec<String>` are formatted as “Name=value” pairs
+    ///
+    /// # Arguments
+    ///
+    /// * domain – the domain of interest. Use `“”` for the default domain.
```

Comment:
Let's not use smart quotes here, they might cause confusion.

Reply:
Editor accident. :/